sqlglot/parser.py

```diff
@@ -877,10 +877,19 @@ def _parse_statement(self) -> t.Optional[exp.Expression]:
             return self.STATEMENT_PARSERS[self._prev.token_type](self)
 
         if self._match_set(Tokenizer.COMMANDS):
+            command_kind = self._prev.text
+            command_body = self._curr.text if self._curr else ""
```

Comment:
why is this part necessary?

Reply:
In the tokenizer, line 847, we have this condition for tokenizing whatever comes after a command as a string:
```python
        if token_type in self.COMMANDS and (
            len(self.tokens) == 1 or self.tokens[-2].token_type == TokenType.SEMICOLON
        ):
```
This works if we have for example a single command to parse, like `SET foo bla bar`, because it will tokenize it as `<COMMAND, text: SET> <STRING, text: foo bla bar>`. The problem with the UDF was that the above condition would never be true because the command appears in the middle of a statement, so we'd tokenize the above like `<COMMAND, text: SET> <VAR, text: foo> <VAR, text: bla> <VAR, text: bar>`, which is harder to parse. The code I added makes sure we consume all remaining tokens and use their texts to construct a string, like it's done for the simple case in the tokenizer.

Reply:
The while loop I added should be a no-op for the general case, because `self._curr` will be `None`.